core/tests/notif_test_wd.c

```diff
@@ -0,0 +1,126 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (c) 2024, Linaro Limited
+ */
+
+#include <initcall.h>
+#include <kernel/callout.h>
+#include <kernel/notif.h>
+#include <kernel/panic.h>
+#include <kernel/tee_time.h>
+#include <kernel/virtualization.h>
+#include <types_ext.h>
+
+#define TEST_WD_TIMER_PERIOD_MS	1000
+
+struct wd_data {
+	bool pending;
+	bool enabled;
+	unsigned int timeout_count;
+	unsigned int call_count;
+	struct callout callout;
+};
+
+static struct wd_data default_wd_data;
+
+static struct wd_data *get_wd_data(void)
+{
+	return &default_wd_data;
+}
+
+static bool test_wd_callback(struct callout *co)
+{
+	struct wd_data *wd = container_of(co, struct wd_data, callout);
+
+	if (wd->pending)
+		wd->timeout_count++;
+	wd->call_count++;
```

Comment:
Here we expect `wd->timeout_count == wd->call_count`, right?
Maybe an test PTA could report to a new xtest test ID whether this is true or not, so that regression test can check the notif and callout frameworks behave as expected.

Reply:
No, we expect `wd->timeout_count` to be `0`, but the PTA thing sounds like a nice extension.